### PR TITLE
Adding AudioComponent and relevant Serialization code and levels with…

### PR DESCRIPTION
**Pull request by pike983**
… audio.

Added AudioComponent, serialization code, level, audio and image assets. Changed Audio source search locations.

### CODE REVIEW

File: Game/Assets/level2.bda [hunk 1]
```diff
@@ -0,0 +1,1123 @@
...
+            "scriptName": "Score"
+          }
+        ],
+        "tag": "text1"
+      },
+      {
+        "tag": "camera",
+        "components": [
+          {
+            "frustumWidth": 1024.0,
```
Comment (Snow106) [inline Game/Assets/level2.bda]: the frustum width for these scenes has to be 1920

File: Engine/src/Engine/SceneBuilder/Components.h [hunk 2]
```diff
@@ -7,6 +7,7 @@
 #include <string>
 #include <functional>
 #include "Engine/Physics/AABB.h"
+#include "../ResourceManagement/Audio.h"
```
Comment (Snow106) [inline Engine/src/Engine/SceneBuilder/Components.h]: don't use relative paths, this lib should be in the includes

File: Engine/src/Engine/ResourceManagement/Audio.cpp [hunk 3]
```diff
@@ -15,8 +15,8 @@ AudioPlayerSingleton* AudioPlayerSingleton::getInstance()
 	if (m_instance == nullptr)
 	{
 		m_instance = new AudioPlayerSingleton();
-		m_engineMediaPath = std::filesystem::current_path().parent_path().string() + "\\Engine\\src\\Media\\";
-		m_gameMediaPath = std::filesystem::current_path().parent_path().string() + "\\Game\\src\\Media\\";
+		m_engineMediaPath = std::filesystem::current_path().parent_path().string() + "\\Engine\\src\\Engine\\Media\\";
```
Comment (Snow106) [inline Engine/src/Engine/ResourceManagement/Audio.cpp:18]: there is a cmake command that involve some of these path; just double check the debug.exe and release.exe are still working properly

**Review by Snow106 — CHANGES_REQUESTED** [review]

File: Game/Assets/level2.bda (hunk 1 above)
Comment (pike983) [reply Game/Assets/level2.bda]: Ok, I'll change that. This was just copied from the original level1.bda from milestone2.

File: Engine/src/Engine/ResourceManagement/Audio.cpp (hunk 3 above)
Comment (pike983) [reply Engine/src/Engine/ResourceManagement/Audio.cpp:18]: Just checked, they do.

**Review by Snow106 — APPROVED** [review]

**Review by KevinAlexV — APPROVED** [review]
Awesome, thanks for implementing this!

**Pull request merged**